crates/autopilot/src/infra/persistence/dto/order.rs

```diff
@@ -272,9 +272,23 @@ pub enum FeePolicy {
     #[serde(rename_all = "camelCase")]
     Surplus { factor: f64, max_volume_factor: f64 },
     #[serde(rename_all = "camelCase")]
+    PriceImprovement {
+        factor: f64,
+        max_volume_factor: f64,
+        quote: Quote,
```

Comment:
I really like that the quote becomes part of the fee policy enum. This makes it very clear how to use it.